### ISSUE TITLE
Resources not freed properly?

### ISSUE THREAD

**Issue by ChristopheBernard** [irmen/Pyro4#142]
I am running a simplistic Pyro4 object server following the doc guidelines, with python 2.7 Windows x64.

The memory footprint is steadily increasing at every request, as if some resources were not released after a connection is closed.

server.py
```
import Pyro4
import socket
import os

@Pyro4.expose
class status_server(object):
    def __init__(self):
        super(status_server, self).__init__()
        self.hostname=socket.gethostname()
        
    def query(self):
        return self.hostname
        
if __name__ == '__main__':
    daemon=Pyro4.Daemon()
    uri=daemon.register(status_server)
    print 'Ready, URI = {}'.format(uri)
    print 'PID = {}'.format(os.getpid())
    
    daemon.requestLoop()
```

pyrorequest.py:

```
import Pyro4
import sys
import time

uri = sys.argv[1]

count=0
while True:
    count+=1
    with Pyro4.Proxy(uri=uri) as px:
        result=px.query()
    print '\rAt step {}'.format(count)
    time.sleep(1)
```

To replicate:

> python server.py

then copy the URI printer at startup

> python pyrorequest.py [URI]

and watch the memory footprint of the server. It increases steadily. The footprint of some servers I had running went up to 3GB after 1-2 days.

**Comment by irmen** [comment]
I can't reproduce your observations so far.  Have been running it for about 10 minutes now (and having decreased the sleep time from 1 to 0.1 sec) and the memory usage of the process is sitting comfortably at a constant 9.540 Kb.     
I've added a ``print(gc.get_objects())`` in the server and can verify that it grows to about 11000 objects, and garbage collects back to about 9500 objects, after which the cycle continues. 
So far, memory use looks constant without leaks. I'll leave it running for some hours and see what it says then.

**Comment by ChristopheBernard** [comment]
Sorry I was misled when I tried to narrow down the problem. The server was collecting uptime data with win32 extensions, and that bit was not releasing resources.

All is clear for me now. I am really sorry for suspecting wrongly Pyro4. Thanks for the great work.

Cheers.

**Comment by irmen** [comment]
Thanks for the explanation; I was about to comment and close the issue myself:
I've just glanced at the server I have kept running for about one day now and at step 823000 it still sits steadily at 12 Mb memory usage.

One thing to point out: it helps a lot if you mention the Pyro version you're using. For instance, in the past few releases I've fixed some obscure resource freeing issues (although they were mostly related with the act of closing down the server, not during regular runtime).
